Framework/API/src/ComponentInfo.cpp

```diff
@@ -13,16 +13,39 @@ namespace API {
 /**
  * Constructor
  * @param componentInfo : Beamline wrapped ComponentInfo
- * @param componentIds : Component Ids ordered by component index
+ * @param componentIds : l-value reference to ComponentIDs ordered by component
+ * index
  */
-ComponentInfo::ComponentInfo(const Beamline::ComponentInfo &componentInfo,
-                             std::vector<Geometry::ComponentID> &&componentIds)
+ComponentInfo::ComponentInfo(
+    const Mantid::Beamline::ComponentInfo &componentInfo,
+    std::vector<Mantid::Geometry::IComponent *> &componentIds)
     : m_componentInfo(componentInfo),
       m_componentIds(
           boost::make_shared<std::vector<Geometry::ComponentID>>(componentIds)),
       m_compIDToIndex(boost::make_shared<
           std::unordered_map<Geometry::IComponent *, size_t>>()) {
+  init();
+}
 
+/**
+ * Constructor
+ * @brief ComponentInfo::ComponentInfo
+ * @param componentInfo
+ * @param componentIds : r-value reference to ComponentIDs ordered by component
+ * index
+ */
+ComponentInfo::ComponentInfo(
+    const Mantid::Beamline::ComponentInfo &componentInfo,
+    std::vector<Mantid::Geometry::IComponent *> &&componentIds)
+    : m_componentInfo(componentInfo),
```

Comment:
@martyngigg I'm not convinced the above is doing what you think. My demo on [wandbox](https://wandbox.org/permlink/72jhJCeYgIxzRcVV)
A slight variant of yours...
```
struct Arg {

    Arg(){std::cout << "default constructor called" << std::endl;};
    Arg(const Arg& ){std::cout << "copy constructor called" << std::endl;}
    Arg(Arg&& ){std::cout << "move constructor called" << std::endl;}
};

struct Foo {
  Foo(const Arg & arg) : m_arg(arg) {
     
  }
  Foo(Arg && arg) : Foo(arg) 
  {} 
    
  /*
  // Uncomment. This avoids the copy!
  Foo(Arg && arg) : m_arg(std::move(arg)) //Foo(arg) 
  {} 
  */

  Arg m_arg;
};

int main() {
    
    Arg arg;
    Foo f1{arg};
    
    Foo f2{Arg{}};
}
```